DataFormats/SoATemplate/test/SoAGenericView_t.cc

```diff
@@ -0,0 +1,191 @@
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#define CATCH_CONFIG_MAIN
+#include <catch.hpp>
+
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+
+GENERATE_SOA_LAYOUT(SoAPositionTemplate,
+                    SOA_COLUMN(float, x),
+                    SOA_COLUMN(float, y),
+                    SOA_COLUMN(float, z),
+                    SOA_SCALAR(int, detectorType))
+
+using SoAPosition = SoAPositionTemplate<>;
+using SoAPositionView = SoAPosition::View;
+using SoAPositionConstView = SoAPosition::ConstView;
+
+GENERATE_SOA_LAYOUT(SoAPCATemplate,
+                    SOA_COLUMN(float, eigenvalues),
+                    SOA_COLUMN(float, eigenvector_1),
+                    SOA_COLUMN(float, eigenvector_2),
+                    SOA_COLUMN(float, eigenvector_3),
+                    SOA_EIGEN_COLUMN(Eigen::Vector3d, candidateDirection))
+
+using SoAPCA = SoAPCATemplate<>;
+using SoAPCAView = SoAPCA::View;
+using SoAPCAConstView = SoAPCA::ConstView;
+
+GENERATE_SOA_LAYOUT(GenericSoATemplate,
+                    SOA_COLUMN(float, xPos),
+                    SOA_COLUMN(float, yPos),
+                    SOA_COLUMN(float, zPos),
+                    SOA_EIGEN_COLUMN(Eigen::Vector3d, candidateDirection))
+
+using GenericSoA = GenericSoATemplate<cms::soa::CacheLineSize::IntelCPU>;
+using GenericSoAView = GenericSoA::View;
+using GenericSoAConstView = GenericSoA::ConstView;
+
+TEST_CASE("SoAGenericView") {
+  // common number of elements for the SoAs
+  const std::size_t elems = 17;
+
+  // buffer sizes
+  const std::size_t positionBufferSize = SoAPosition::computeDataSize(elems);
+  const std::size_t pcaBufferSize = SoAPCA::computeDataSize(elems);
+
+  // memory buffer for the SoA of positions
+  std::unique_ptr<std::byte, decltype(std::free) *> bufferPos{
+      reinterpret_cast<std::byte *>(aligned_alloc(SoAPosition::alignment, positionBufferSize)), std::free};
+  // memory buffer for the SoA of the PCA
+  std::unique_ptr<std::byte, decltype(std::free) *> bufferPCA{
+      reinterpret_cast<std::byte *>(aligned_alloc(SoAPCA::alignment, pcaBufferSize)), std::free};
+
+  // SoA Layouts
+  SoAPosition position{bufferPos.get(), elems};
+  SoAPCA pca{bufferPCA.get(), elems};
+
+  // SoA Views
+  SoAPositionView positionView{position};
+  SoAPositionConstView positionConstView{position};
+  SoAPCAView pcaView{pca};
+  SoAPCAConstView pcaConstView{pca};
+
+  // fill up
+  for (size_t i = 0; i < elems; i++) {
+    positionView.x()[i] = static_cast<float>(i);
+    positionView.y()[i] = static_cast<float>(i) * 2.0f;
+    positionView.z()[i] = static_cast<float>(i) * 3.0f;
```

Comment:
I would slightly prefer
```suggestion
    positionView[i].x() = static_cast<float>(i);
    positionView[i].y() = static_cast<float>(i) * 2.0f;
    positionView[i].z() = static_cast<float>(i) * 3.0f;
```

You could also simplify it a bit as
```suggestion
    positionView[i] = { i * 1.0f, i * 2.0f, i * 3.0f };
```